Wikipedia/Code/WMFTableOfContentsPresentationController.swift

```diff
@@ -107,9 +128,32 @@ public class WMFTableOfContentsPresentationController: UIPresentationController
     
     override public func viewWillTransitionToSize(size: CGSize, withTransitionCoordinator transitionCoordinator: UIViewControllerTransitionCoordinator) {
         super.viewWillTransitionToSize(size, withTransitionCoordinator: transitionCoordinator)
-        
+
         transitionCoordinator.animateAlongsideTransition({(context: UIViewControllerTransitionCoordinatorContext!) -> Void in
             self.backgroundView.frame = self.containerView!.bounds
             }, completion:nil)
+
+        if(UIDeviceOrientationIsLandscape(UIDevice.currentDevice().orientation))
```

Comment:
:-1: do this logic in `willTransitionToTraitCollection` and check `newCollection.verticalSizeClass` for either `.Regular` or `.Compact`